Hivemind/Assets/Scripts/Minds/Gathering.cs

```diff
@@ -165,11 +172,24 @@ public void Execute(Ant ant)
         }
     }
 
+    private void FoundResource()
```

Comment:
Just put everything in the Discover function and start the coroutine where this function is currently called